### PR TITLE
Lockless changes for dedicated procedure

### PR DESCRIPTION
**Pull request by rsarwad**
Summary: Lock less changes for dedicated procedure
Type: Refactor
Test Plan: Executed s1ap tester test suite and newly added dedicated bearer  and multi pdn test cases

### CODE REVIEW

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 1]
```diff
@@ -2478,91 +2487,88 @@ void mme_app_handle_modify_ue_ambr_request(mme_app_desc_t *mme_app_desc_p,
...
-  itti_send_msg_to_task(TASK_NAS_MME, INSTANCE_DEFAULT, message_p);
+  if ((nas_proc_create_dedicated_bearer(&activate_ded_bearer_req)) !=
+    RETURNok) {
+    OAILOG_ERROR(
+      LOG_MME_APP,
+      "Failed to handle bearer activation at NAS module for ue_id "
+      MME_UE_S1AP_ID_FMT "\n", ue_context_p->mme_ue_s1ap_id);
+  }
+  if(activate_ded_bearer_req.tft) {
+    free(activate_ded_bearer_req.tft);
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c]: change to free_wrapper()

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 2]
```diff
@@ -753,9 +771,15 @@ void mme_app_handle_delete_session_rsp(mme_app_desc_t *mme_app_desc_p,
 //------------------------------------------------------------------------------
 {
   struct ue_mm_context_s *ue_context_p = NULL;
+  emm_cn_pdn_disconnect_rsp_t pdn_disconnect_rsp = {0};
 
   OAILOG_FUNC_IN(LOG_MME_APP);
-  DevAssert(delete_sess_resp_pP);
+  if (!delete_sess_resp_pP) {
+    OAILOG_DEBUG(
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c]: Change to the log level to error

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 3]
```diff
@@ -2931,202 +2942,124 @@ void mme_app_handle_path_switch_request(mme_app_desc_t *mme_app_desc_p,
...
-      "We didn't find this mme_ue_s1ap_id in list of UE: " MME_UE_S1AP_ID_FMT
-      "\n",
+      "ERAB released successfully at UE with ERAB-ID:%u for "
+      "ue_id" MME_UE_S1AP_ID_FMT "\n",
+      e_rab_id,
       e_rab_rel_rsp->mme_ue_s1ap_id);
-    OAILOG_FUNC_OUT(LOG_MME_APP);
   }
 
   for (int i = 0; i < e_rab_rel_rsp->e_rab_rel_list.no_of_items; i++) {
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c]: Correct the condition

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 4]
```diff
@@ -2931,202 +2942,124 @@ void mme_app_handle_path_switch_request(mme_app_desc_t *mme_app_desc_p,
...
-    OAILOG_ERROR(
-      LOG_MME_APP,
-      "No bearer context found ue " MME_UE_S1AP_ID_FMT " ebi %u\n",
-      itti_erab_rel_cmd->ue_id,
-      itti_erab_rel_cmd->ebi);
+    s1ap_e_rab_rel_cmd->e_rab_to_be_rel_list.no_of_items = 1;
+    s1ap_e_rab_rel_cmd->e_rab_to_be_rel_list.item[0].e_rab_id =
+      bearer_context->ebi;
   }
+  //s1ap_e_rab_rel_cmd->e_rab_to_be_rel_list.item[0].cause = 0; //Pruthvi TDB
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c]: Make it "TODO"

File: lte/gateway/c/oai/common/itti_free_defined_msg.c [hunk 5]
```diff
@@ -160,6 +159,14 @@ void itti_free_msg_content(MessageDef* const message_p)
     case S1AP_UE_CONTEXT_RELEASE_COMPLETE:
       // DO nothing
       break;
+    case S1AP_E_RAB_REL_CMD:
+      bdestroy_wrapper(&message_p->ittiMsg.s1ap_e_rab_rel_cmd.nas_pdu);
+      break;
+    case S1AP_E_RAB_SETUP_REQ:
+      bdestroy_wrapper(
+        &message_p->ittiMsg.s1ap_e_rab_setup_req.e_rab_to_be_setup_list.item[0]
```
Comment (ulaskozat) [inline lte/gateway/c/oai/common/itti_free_defined_msg.c:167]: should not this free all the items? Is it guaranteed that only one item is present?

File: lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c [hunk 6]
```diff
@@ -1418,10 +1464,12 @@ void mme_app_handle_e_rab_setup_rsp(mme_app_desc_t *mme_app_desc_p,
        i++) {
     e_rab_id_t e_rab_id =
       e_rab_setup_rsp->e_rab_failed_to_setup_list.item[i].e_rab_id;
-    bearer_context_t *bc =
+    bearer_context_t* bc =
       mme_app_get_bearer_context(ue_context_p, (ebi_t) e_rab_id);
     if (bc->bearer_state & BEARER_STATE_SGW_CREATED) {
-      //S1ap_Cause_t cause = e_rab_setup_rsp->e_rab_failed_to_setup_list.item[i].cause;
+      /* S1ap_Cause_t cause =
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c]: Remove commented code or add a TODO if this is to be included in future

File: lte/gateway/c/oai/common/itti_free_defined_msg.c (hunk 5 above)
Comment (rsarwad) [reply lte/gateway/c/oai/common/itti_free_defined_msg.c:167]: Yes only one is present. In fun, mme_app_handle_erab_setup_req(), itti message is allocated and filled with required information to be sent in "e_rab_set_up_request" message to eNB.
In itti message only one nas pdu is filled. So we are freeing only allocated memory

Comment (ulaskozat) [reply lte/gateway/c/oai/common/itti_free_defined_msg.c:167]: For now it is okay, but this is supposed to be a generic call for freeing message content.

Comment (ulaskozat) [reply lte/gateway/c/oai/common/itti_free_defined_msg.c:167]: We should make cleaning stage compatible with the message definition to make this future proof.

Comment (rsarwad) [reply lte/gateway/c/oai/common/itti_free_defined_msg.c:167]: Yes you are right, we need to clean code aligned with message definition.
Will make this change as part new PR

**Pull request closed without merging**